lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -290,19 +293,19 @@ data Listen
 
 -- | Start the application server, using the given settings and a bound socket.
 start
-    :: forall t .
+    :: forall t (n :: NetworkDiscriminant).
         ( Buildable (ErrValidateSelection t)
         , DefineTx t
-        , DecodeAddress t
-        , EncodeAddress t
-        , KeyToAddress t RndKey
-        , KeyToAddress t SeqKey
+        , DecodeAddress n
+        , EncodeAddress n
+        , PaymentAddress n ByronKey
+        , PaymentAddress n ShelleyKey
         )
     => Warp.Settings
     -> Trace IO Text
     -> Socket
-    -> ApiLayer (RndState t) t RndKey
-    -> ApiLayer (SeqState t) t SeqKey
+    -> ApiLayer (RndState n) t ByronKey
+    -> ApiLayer (SeqState n) t ShelleyKey
```

Comment:
@paweljakubas now the relationship between era, keys and backend is made more obvious and here, we clearly see what we support. Yoroi is missing as being a combination of `SeqState` and `ByronKey`